airbyte-integrations/connectors/source-tiktok-marketing/source_tiktok_marketing/components/advertiser_ids_partition_router.py

```diff
@@ -0,0 +1,61 @@
+# Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+
+import json
+from typing import Any, Iterable, Mapping
+
+import dpath.util
+from airbyte_cdk.sources.declarative.partition_routers.substream_partition_router import SubstreamPartitionRouter
+from airbyte_cdk.sources.declarative.types import StreamSlice
+
+
+class MultipleAdvertiserIdsPartitionRouter(SubstreamPartitionRouter):
+    """
+    Custom AdvertiserIdsPartitionRouter and AdvertiserIdPartitionRouter partition routers are used to get advertiser_ids
+    as slices for streams where it uses as request param.
+
+    When user uses sandbox account it's impossible to get advertiser_ids via API.
+    In this case user need to provide advertiser_id in a config and connector need to use provided ids
+    and do not make requests to get this id.
+
+    When advertiser_id not provided components get slices as usual.
+    Main difference between AdvertiserIdsPartitionRouter and AdvertiserIdPartitionRouter is
+    that AdvertiserIdPartitionRouter returns multiple advertiser_ids in a one slice when id is not provided,
+    e.g. {"advertiser_ids": '["11111111", "22222222"]', "parent_slice": {}}.
+    And AdvertiserIdPartitionRouter returns single slice for every advertiser_id as usual.
```

Comment:
> I think one of these should reference AdvertiserIdsPartitionRouter.

updated with correct names.

> I think we should try to avoid multiple advertiser_ids in a single slice if we can avoid it. Couldn't we just read in the config's advertiser ids and return multiple slices?

`MultipleAdvertiserIdPartitionRouter` is used only in one stream Advertisers to fetch more then one advertisers at once, reduce amount of requests. It's how it was implemented in py version of the connector. All other streams use `SingleAdvertiserIdPartitionRouter`. ` advertiser ids ` is not required property in the config so we can't avoid reading parent stream.